### PR TITLE
Add NTP search promo P3A question

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves brave/brave-browser#24065

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Regression testing of brave/brave-browser#23191 recommended.

1. Launch browser using `--enable-features="BraveSearchNTP"` switch, with fresh profile.
2. Access `brave://local-state`, ensure `Brave.Search.Promo.NewTabPage` does not exist.
3. Access `brave://settings/search`, change search engine to something other than Brave.
4. Ensure sponsored images are disabled and access new tab page, ensure promo appears.
5. Check local state, the metric should now exist, and should be set to 0.
6. Make search query via NTP promo.
7. Metric should have a value of 1.
8. Change the default search engine to Brave by either clicking on the prompt that appears on the search engine results page, or via `brave://settings/search` (may be good to test both).
9. Relevant metric should have a value of 3.
10. Start over with a fresh profile, see promo but do not perform a search.
11. Change search engine to Brave via one of the methods in step 8. (Accessing https://search.brave.com/search?q=test&action=makeDefault directly will force the SERP prompt to show up.)
12. Relevant metric should have a value of 2.

### CODE REVIEW

**Review by rillian — APPROVED** [review]

File: browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc [hunk 1]
```diff
@@ -83,6 +86,10 @@ void BraveNewTabPageHandler::InitForSearchPromotion() {
                           base::Unretained(this)));
   template_url_service_observation_.Observe(
       TemplateURLServiceFactory::GetForProfile(profile_));
+
+  brave_search_conversion::p3a::RecordPromoShown(
```
Comment (simonhong) [inline browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc:90]: This is called whenever NTP is created if promotion is enabled.
However, NTP will not show promotion when sponsored images are displayed.
When sponsored images are displayed, this should not be called.
I'll push another commit.

Comment (simonhong) [reply browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc:111]: Pushed https://github.com/brave/brave-core/pull/14189/commits/0e00a36fca72a75a57773c5e6defe7e1496104c6

**Review by simonhong — APPROVED** [review]
++ first commit :)

**Comment by simonhong** [comment]
Kindly ping @iefremov  :)

**Pull request merged**